### PR TITLE
Fixes button states

### PR DESCRIPTION
**Pull request by ashfurrow**
The bid button now displays the correct button state, including the "highest bidder" state. Users are notified when they're outbid. The bid button's opacity is no longer affected by disabling it.

### CODE REVIEW

File: Artsy_Tests/View_Controller_Tests/Live_Auction/LiveAuctionBidViewControllerTests.swift [hunk 1]
```diff
@@ -79,13 +79,13 @@ class LiveAuctionPlaceMaxBidViewControllerSpecs: QuickSpec {
...
-                "in progress": .BiddingInProgress,
-                "is max bidder": .BidSuccess(isMaxBidder: true),
-                "not max bidder": .BidSuccess(isMaxBidder: false),
-                "network issues": .BidNetworkFail,
-                "waiting": .LotWaitingToOpen,
-                "sold": .LotSold,
+            let examples:[String: [LiveAuctionBiddingProgressState]] = [
+                "in progress": [.BiddingInProgress],
+                "is max bidder": [.BidSuccess],
+//                "not max bidder": [.BidSuccess, .Biddable(askingPrice: 1_000_00, currencySymbol: "¥")],
```
Comment (ashfurrow) [inline Artsy_Tests/View_Controller_Tests/Live_Auction/LiveAuctionBidViewControllerTests.swift]: Left this in to make it easier to pick up when we finish implementing the controller being tested here.

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
        
            2 Warnings
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Needs testing on a Phone if change is non-trivial</td>
    </tr>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
    
    
  </tbody>
</table>
  

  



Time | Class | Function |
| --- | ----- | ----- |245.2ms | [AuctionViewController.swift](/artsy/eigen/tree/ashfurrow-events/Artsy/View_Controllers/Auction/AuctionViewController.swift#L181) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
205.5ms | [SwiftyJSON.swift](/artsy/eigen/tree/ashfurrow-events/Pods/SwiftyJSON/Source/SwiftyJSON.swift#L445) | get {}
168.5ms | [WebSocket.swift](/artsy/eigen/tree/ashfurrow-events/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)
100.1ms | [Observable.swift](/artsy/eigen/tree/ashfurrow-events/Pods/Interstellar/Sources/Observable.swift#L71) | private final func nextTokenHash() -> Int
71.0ms | [LiveAuctionsAdminViewController.swift](/artsy/eigen/tree/ashfurrow-events/Artsy/View_Controllers/Live_Auctions/LiveAuctionsAdminViewController.swift#L22) | @objc override func viewDidLoad()
64.1ms | [AuctionLotMetadataStackScrollView.swift](/artsy/eigen/tree/ashfurrow-events/Artsy/View_Controllers/Live_Auctions/Views/AuctionLotMetadataStackScrollView.swift#L18) | required init(viewModel: LiveAuctionLotViewModelType, sideMargin: String)
61.7ms | [RefinementOptionsViewController+Private.swift](/artsy/eigen/tree/ashfurrow-events/Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift#L63) | func stackView() -> ORStackView




<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

**Comment by orta** [comment]
Merging this, as it goes into #1602

**Comment by devangt** [comment]
Excited for this one!! 

-devang




On Fri, Jun 3, 2016 at 2:58 PM -0700, "Orta" <notifications@github.com> wrote:












Merged #1605.

**Pull request merged**